File: Programowanie_Obiektowe/K2-Probne/Zestaw_1/Stadium.java
package pl.uwm.wmii.kaz.lemmgator;

public class Stadium extends Court {

    private String name;
    private String common_name;
    private int capacity;

    public Stadium(double width, double length, String address, int year_built, String name, String common_name, int capacity) {
        super(width, length, address, year_built);
        this.name = name;
        this.common_name = common_name;
        if (capacity < 0) capacity = 0;
        this.capacity = capacity;
    }

    public Stadium(double width, double length, String address, int year_built, String name, int capacity) {
        this(width, length, address, year_built, name, "", capacity);
    }

    public Stadium(String address, int year_built, String name) {
        this(68, 100, address, year_built, name, "", 0);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCommon_name() {
        return common_name;
    }

    public void setCommon_name(String common_name) {
        this.common_name = common_name;
    }

    public int getCapacity() {
        return capacity;
    }

    public void setCapacity(int capacity) {
        if (capacity < 0) capacity = 0;
        this.capacity = capacity;
    }

    @Override
    public String toString() {
        if (common_name == "") {
            return super.toString() + "\n"
                    + "Nazwa: " + getName() + ".\n"
                    + "Pojemnosc stadionu: " + getCapacity() + ".";
        } else {
            return super.toString() + "\n"
                    + "Nazwa: " + getName() + ".\n"
                    + "Nazwa zwyczajowa: " + getCommon_name() + ".\n"
                    + "Pojemnosc stadionu: " + getCapacity() + ".";
        }
    }
}

File: Programowanie_Obiektowe/K1-Probne/stadium.py
from court import Court


class Stadium(Court):
    __name: str
    __common_name: str
    __capacity: int

    def __init__(self, width: float, length: float,
                 address: str, year_built: int, name: str,
                 common_name: str = '', capacity: int = 0) -> None:
        super().__init__(width, length, address, year_built)
        self.__name = name
        self.__common_name = common_name
        if capacity >= 0:
            self.__capacity = capacity
        else:
            self.__capacity = 0

    @property
    def name(self) -> str:
        return self.__name

    @property
    def common_name(self) -> str:
        return self.__common_name

    @property
    def capacity(self) -> int:
        return self.__capacity

    @name.setter
    def name(self, value: str) -> None:
        self.__name = value

    @common_name.setter
    def common_name(self, value: str) -> None:
        self.__common_name = value

    @capacity.setter
    def capacity(self, value: int) -> None:
        if value < 0:
            print('Attribute error')
        else:
            self.__capacity = value

    def __eq__(self, other: 'Stadium') -> bool:
        return self.capacity == other.capacity and \
               self.area() == other.area()

    def __ne__(self, other: 'Stadium') -> bool:
        return self.capacity != other.capacity or \
               self.area() != other.area()

    def __str__(self) -> str:
        if self.__common_name != '':
            return f'Boisko wybudowane w roku ' \
                   f'{self.year_built}, o długości ' \
                   f'{self.length} metrów i szerokości {self.width} metrów.' \
                   f'\nPole powierzchni: {self.area()} mkw.' \
                   f'\nAdres: {self.address}.' \
                   f'\nNazwa: {self.name}.' \
                   f'\nPojemność stadionu: {self.capacity} osób.'
        else:
            return f'Boisko wybudowane w roku ' \
                   f'{self.year_built}, o długości ' \
                   f'{self.length} metrów i szerokości {self.width} metrów.' \
                   f'\nPole powierzchni: {self.area()} mkw.' \
                   f'\nAdres: {self.address}.' \
                   f'\nNazwa: {self.name}.' \
                   f'\nNazwa zwyczajowa: {self.common_name}.' \
                   f'\nPojemność stadionu: {self.capacity} osób.'
